src/Utils/Rotation.cpp

```diff
@@ -312,15 +312,335 @@ void utils::Rotation::checkUnitary()
+                // for the two large singular values
+                if(!done)
+                {
+                    for(int dim = 0; dim < 3; dim++)
+                    {
+                        int dim_a = dim;
+                        int dim_b = (dim + 1) % 3;
+                        int dim_c = (dim + 2) % 3;
+
+                        if(lambda(dim_a) < tolerance)
+                        {
+                            // columns dimB and dimC are both good,
+                            // but column dimA corresponds to a tiny singular value
+
+                            utils::Vector3d vec1(Um(0,dim_b), Um(1,dim_b), Um(2,dim_b)); // column dimB
+                            utils::Vector3d vec2(Um(0,dim_c), Um(1,dim_c), Um(2,dim_c)); // column dimC
+                            utils::Vector3d vec3 = (vec1.cross(vec2)).normalized();
+                            Um(0, dim_a) = vec3(0);
+                            Um(1, dim_a) = vec3(1);
+                            Um(2, dim_a) = vec3(2);
+                            if(Um.determinant() < 0.0)
+                            {
+                                Um(0, dim_a) *= -1.0;
+                                Um(1, dim_a) *= -1.0;
+                                Um(2, dim_a) *= -1.0;
+                            }
+                            done = 1;
+                            break; // out of for
+                        }
+                    }
+                }
+
+                if ((!done) && (modified_SVD == 1))
+                {
+                    // Handle situation:
+                    // negative determinant (Tet is inverted in solid mechanics)
+                    //    - check if det(U) == -1
+                    //    - If yes, then negate the minimal element of Sigma
+                    //      and the corresponding column of U
+
+                    float det_U = Um.determinant();
+                    if (det_U < 0.0)
+                    {
+                        // negative determinant
+                        // find the smallest singular value (they are all non-negative)
+                        int smallest_singular_value_idx = 0;
+                        for(int dim=1; dim<3; dim++)
+                            if (lambda(dim) < lambda(smallest_singular_value_idx))
+                                smallest_singular_value_idx = dim;
+
+                        // negate the smallest singular value
+                        lambda(smallest_singular_value_idx) *= -1.0;
+                        Um(0, smallest_singular_value_idx) *= -1.0;
+                        Um(1, smallest_singular_value_idx) *= -1.0;
+                        Um(2, smallest_singular_value_idx) *= -1.0;
+                    }
+                }
+            }
+            R = Um * Vm.transpose();
+        }
+        else
+        {
+            // R = Mk^T
+            R = Mk.transpose();
+        }
+
+
+        std::cout << "On s'est rendu, youpi!\n";
```

Comment:
Please note that the prefered return line in `std::cout` is not `\n` but `std::endl` (just a piece of trivia since this french message is probably going to be removed anyway :P)